src/TorchSharp/NN/Module.cs

```diff
@@ -1083,7 +1084,7 @@ protected virtual void RegisterComponents()
 
                     foreach (var field in GetType().GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)) {
 
-                        var fieldName = field.Name;
+                        var fieldName = field.GetCustomAttribute<ComponentNameAttribute>()?.Name ?? field.Name;
```

Comment:
There are 3 other places where the code goes through the fields and does stuff, specifically, when moving a module to/from GPU, or converting to another element type. They need the same treatment.